lib/mempool/composedFees.js

```diff
@@ -0,0 +1,246 @@
+const FeeEstimator = require('../covenants/fees');
+const rules = require('../covenants/rules');
+
+/**
+ * Composed Estimator
+ * 
+ * There are four limits for how many transactions can go in a block, each of which could fill up separately:
+ *  - The main fee market for block space, limited by consensus.MAX_BLOCK_WEIGHT
+ *  - A submarket for FINALIZE, REGISTER, and RENEW covenants, limited by consensus.MAX_BLOCK_RENEWALS
+ *  - A submarket for OPEN, UPDATE, TRANSFER and REVOKE covenants, limited by consensus.MAX_BLOCK_UPDATES
+ *  - A submarket for  OPEN covenants, limited by consensus.MAX_BLOCK_OPENS. Subset of the previous limit.
+
+ * @alias module:mempool.ComposedEstimator
+ */
+
+class ComposedEstimator extends bio.Struct {
+  /**
+   * Create an estimator.
+   * @constructor
+   * @param {Logger?} logger
+   */
+
+  constructor(logger) {
+    super();
+
+    this.blockspaceFees = new FeeEstimator(logger);
+    this.openFees = new FeeEstimator(logger);
+    this.updateFees = new FeeEstimator(logger);
+    this.renewalFees = new FeeEstimator(logger);
+  }
+
+  /**
+   * Initialize the estimator.
+   * @private
+   */
+
+  init() {
+    this.blockspaceFees.init();
+    this.openFees.init();
+    this.updateFees.init();
+    this.renewalFees.init();
+  }
+
+  /**
+   * Reset the estimator.
+   */
+
+  reset() {
+    this.blockspaceFees.reset();
+    this.openFees.reset();
+    this.updateFees.reset();
+    this.renewalFees.reset();
+  }
+
+  /**
+   * Stop tracking a tx. Remove from map.
+   * @param {Hash} hash
+   */
+
+  removeTX(hash) {
+    this.blockspaceFees.removeTX(hash);
+    this.openFees.removeTX(hash);
+    this.updateFees.removeTX(hash);
+    this.renewalFees.removeTX(hash);
+  }
+
+  /**
+   * Test whether a fee should be used for calculation.
+   * @param {Amount} fee
+   * @param {Number} priority
+   * @returns {Boolean}
+   */
+
+  isFeePoint(fee, priority) {
+    return this.blockspaceFees.isFeePoint(fee, priority);
+  }
+
+  /**
+   * Test whether a priority should be used for calculation.
+   * @param {Amount} fee
+   * @param {Number} priority
+   * @returns {Boolean}
+   */
+
+  isPriPoint(fee, priority) {
+    return this.blockspaceFees.isPriPoint(fee, priority);
+  }
+
+  /**
+   * Process a mempool entry.
+   * @param {MempoolEntry} entry
+   * @param {Boolean} current - Whether the chain is synced.
+   */
+
+  processTX(entry, current) {
+    this.blockspaceFees.processTX(entry, current);
+    if (rules.countOpens(entry.tx) > 0) this.openFees.processTX(entry, current);
```

Comment:
Nit: for one line if statements, always place on next line with no curly braces

I won't comment on this again since there are a bunch of different places.